imap_processing/tests/mag/test_mag_decom.py

```diff
@@ -35,15 +36,49 @@ def test_mag_decom():
     assert len(l0) == len(expected_output.index)
 
 
-def test_mag_raw_cdf():
+def test_mag_raw_xarray():
     current_directory = Path(__file__).parent
     burst_test_file = current_directory / "mag_l0_test_data.pkts"
     l0 = decom_packets(str(burst_test_file))
 
-    output_data = export_to_xarray(l0)
+    norm_data, burst_data = export_to_xarray(l0)
     required_attrs = list(
         global_attrs.GlobalInstrumentAttrs("", "", "").output().keys()
     )
 
-    assert all([item in list(output_data.attrs.keys()) for item in required_attrs])
-    assert all([item is not None for _, item in output_data.attrs.items()])
+    assert all([item in list(norm_data.attrs.keys()) for item in required_attrs])
+    assert all([item is not None for _, item in norm_data.attrs.items()])
+
+    assert all([item in list(burst_data.attrs.keys()) for item in required_attrs])
+    assert all([item is not None for _, item in burst_data.attrs.items()])
+
+    expected_norm_len = 17
+    print(norm_data.dims["Epoch"])
+    assert norm_data.dims["Epoch"] == expected_norm_len
+
+    expected_burst_len = 19
+    assert burst_data.dims["Epoch"] == expected_burst_len
+
+
+def test_mag_raw_cdf_generation(tmp_path):
+    current_directory = Path(__file__).parent
+    test_file = current_directory / "mag_l0_test_data.pkts"
+    l0 = decom_packets(str(test_file))
+
+    norm_data, burst_data = export_to_xarray(l0)
+
+    test_data_path_norm = tmp_path / "mag_l1a_raw-normal_20210101_20210102_v01-01.cdf"
+
+    output = write_cdf(norm_data, test_data_path_norm)
+    assert Path.exists(test_data_path_norm)
+
+    input_xarray = cdf_to_xarray(output)
+    assert input_xarray.attrs.keys() == norm_data.attrs.keys()
+
+    test_data_path_burst = tmp_path / "mag_l1a_raw-burst_20210101_20210102_v01-01.cdf"
+
+    output = write_cdf(burst_data, test_data_path_burst)
+    assert Path.exists(test_data_path_burst)
```

Comment:
Might want to assert it doesn't exist before the functions as well? (Similar above this)
```suggestion
    assert test_data_path_burst.exists()
```